Gems/PhysX/Code/Tests/ColliderScalingTests.cpp

```diff
@@ -54,24 +57,13 @@ namespace PhysXEditorTests
 
     TEST_F(PhysXEditorFixture, BoxCollider_NonUniformScale_RuntimePhysicsAabbCorrect)
     {
-        EntityPtr editorEntity = CreateInactiveEditorEntity("Box");
-
-        Physics::ColliderConfiguration colliderConfig;
-        colliderConfig.m_rotation = AZ::Quaternion::CreateRotationZ(AZ::DegToRad(45.0f));
-        colliderConfig.m_position = AZ::Vector3(1.0f, 2.0f, 3.0f);
-        Physics::BoxShapeConfiguration boxShapeConfig(AZ::Vector3(0.5f, 0.7f, 0.9f));
-        editorEntity->CreateComponent<PhysX::EditorColliderComponent>(colliderConfig, boxShapeConfig);
-        editorEntity->CreateComponent<AzToolsFramework::Components::EditorNonUniformScaleComponent>();
-        editorEntity->CreateComponent<PhysX::EditorStaticRigidBodyComponent>();
-        editorEntity->Activate();
-
-        AZ::EntityId editorId = editorEntity->GetId();
-        AZ::Transform worldTM;
-        worldTM.SetUniformScale(1.5f);
-        worldTM.SetTranslation(AZ::Vector3(5.0f, 6.0f, 7.0f));
-        worldTM.SetRotation(AZ::Quaternion::CreateRotationX(AZ::DegToRad(30.0f)));
-        AZ::TransformBus::Event(editorId, &AZ::TransformBus::Events::SetWorldTM, worldTM);
-        AZ::NonUniformScaleRequestBus::Event(editorId, &AZ::NonUniformScaleRequests::SetScale, AZ::Vector3(0.7f, 0.9f, 1.1f));
+        const AZ::Vector3 boxDimensions(0.5f, 0.7f, 0.9f);
+        const AZ::Transform transform(AZ::Vector3(5.0f, 6.0f, 7.0f), AZ::Quaternion::CreateRotationX(AZ::DegToRad(30.0f)), 1.5f);
+        const AZ::Vector3 translationOffset(1.0f, 2.0f, 3.0f);
+        const AZ::Quaternion rotationOffset = AZ::Quaternion::CreateRotationZ(AZ::DegToRad(45.0f));
+        const AZ::Vector3 nonUniformScale(0.7f, 0.9f, 1.1f);
+
+        EntityPtr editorEntity = CreateBoxPrimitiveColliderEditorEntity(boxDimensions, transform, translationOffset, rotationOffset, nonUniformScale);
```

Comment:
This pattern is repeated several times within these tests which indicates that this would be a good candidate for a refactor into a test basis or another test helper to help prevent code duplication.

Reply:
This PR is already removing lot of boiler plate by using the `CreateXXXXColliderEditorEntity` helpers, so it's a step into the right direction, unfortunately now it could look like even more refactoring can be done sine new patterns emerge, but I think we can continue improving the code in subsequent PRs. 

For this particular one that's highlighted in this file, each shape has their own values and there are 2 tests with that "pattern" for each shape (except cylinder that has more, but again, different data for each test), so it's not a lot of repetition since the data is different for each shape. So unless we go for a bigger refactor of these tests to try to make the same tests cover all the shapes, which I'm not sure it's something I'd go for since for unit tests I'd aim for simplicity to make them easy to understand. It's always a balance, but I think the tests on this file have a nice length and pattern at the moment IMO.